### ISSUE TITLE
Empty (Invisible) Header Item when Using Sticky Headers

### ISSUE THREAD

**Issue by nluetkemeyer** [davideas/FlexibleAdapter#297]
I think I've found a small bug - which I was able to reproduce on your sample app. In my project, I'm currently using the SNAPSHOT version and displaying it on a Samsung Galaxy using API 23.

To put it simply, I have created a list with multiple headers, similar to "Headers and Sections" in your app. I've been using your library for a while and finally got around to testing different scenarios, so I don't really know how long this has been a problem.

If you scroll the list in the way I explain below, one header will initially be invisible when it enters the screen from the bottom. This header will remain invisible until another header makes the "sticky" transition.

Steps are as follows:
1. Open "Headers and Sections" in sample app
2. Scroll down a small amount - to the bottom of "Simple Item 2", for instance
3. Scroll back up, just enough to trigger a new header to be created, but still invisible - to the upper half of "Simple Item 1" (NOTE: Do not scroll all the way back to the top)
4. Now scroll back down, and you should soon see a header that is initially invisible ("Header 4" in my case) and it will remain invisible until Header 3 becomes "sticky".

This is easier to recognize when you put many section items under "Header 1" (about 2x screen height - in my case, 15 Simple Items) by dragging down other headers. If you repeat the steps above, you will notice that Header 2 is initially invisible when it enters the screen and will not become visible until you scroll it enough to become "sticky".

Let me know if something is unclear or if you are unable to reproduce this. I am happy to send you a video or something.

**Comment by davideas** [comment]
@nluetkemeyer, yes better a video. But does this occur in rc1?

**Comment by nluetkemeyer** [comment]
@davideas I've compiled rc1 in my project and it looks like the same thing occurs. Here is a link to the video of the situation...

https://www.dropbox.com/s/j5mh9cznxygwxxw/empty_header_flexible_adapter.mp4?dl=0

**Comment by davideas** [comment]
Thanks for the video, I will work on it. I know what is causing that now.

**Comment by davideas** [comment]
@nluetkemeyer, for me should be resolved. Would you check?

**Comment by nluetkemeyer** [comment]
@davideas yes, that worked! Thanks.

I just have a question, so I won't open another issue. I've created a table using HorizontalScrollViews in each of my headers and items (with vertical RecyclerView). Using a simple listener, I sync and scroll them all at once when any one of them are scrolled individually. 

Problem is, all positions less than scroll position have some lag or delay when scrolling. This happens no matter what order I scroll the views in. I'm not exactly sure the source of the issue, but I just wanted to check if you had any idea before I went any deeper... maybe animators? I have attached another video. The delay is slight, but I think it should be noticeable.

https://www.dropbox.com/s/3lp4ev1rmjaqdoz/HorizontalScrollView.mp4?dl=0
